packages/nft/src/streams.ts

```diff
@@ -0,0 +1,36 @@
+import * as eventsErc721 from "./eventsErc721";
+import * as eventsCryptoPunks from "./eventsCryptoPunks";
+import { EventStreamSchema, serializers } from "@proximaone/stream-schema-base";
+import { ChainReference } from "./model";
+
+export type Erc721StreamEvent =
+  | eventsErc721.NewNftCollection
+  | eventsErc721.Approval
+  | eventsErc721.ApprovalForAll
+  | eventsErc721.Transfer & {
+  ref?: ChainReference
+};
+
+export type CryptoPunksEvent =
+  | eventsCryptoPunks.Assign
+  | eventsCryptoPunks.FundTransfer
+  | eventsCryptoPunks.PunkTransfer
+  | eventsCryptoPunks.PunkOffered
+  | eventsCryptoPunks.PunkBidEntered
+  | eventsCryptoPunks.PunkBidWithdrawn
+  | eventsCryptoPunks.PunkBought
+  | eventsCryptoPunks.PunkNoLongerForSale & {
+  ref?: ChainReference
+};
+
+export const erc721: EventStreamSchema<Erc721StreamEvent> = {
```

Comment:
Events look good, but let's remove erc721 from the name. Idea of this package is just to hold "generalized" version of nft stream which can be used for events from other blockchains also.

So I'm fine with just `nft` name;

CryptoPunks also should not be there. They can be scoped in the CryptoPunk streaming app by now.

Finally all these streams will look like:


ETH BLOCKCHAIN -> ERC721 Discovery -> NFT Stream 1
ETH BLOCKCHAIN -> CryptoPunk Parse -> CryptoPunk Event stream -> NFT STREAM 2

NFT STREAM = MERGE(NFT STREAM 1, NFT STREAM 2)